sites/funmooc/src/backend/base/views.py

```diff
@@ -0,0 +1,41 @@
+"""Views for the fun-mooc site."""
+from django.conf import settings
+from django.db.models import Q
+from django.http import HttpResponsePermanentRedirect
+
+from cms.api import Page
+from cms.constants import PUBLISHER_STATE_PENDING
+from richie.apps.core.views.error import error_view_handler
+
+
+def redirect_edx_courses(request, organization, course, session):
+    """
+    The richie site is hosted on the same domain as OpenEdX before.
+    Redirect OpenEdX course urls to the corresponding Richie course urls.
+    """
+
+    def get_redirect_url(**kwargs):
+        """Look for a published page matching the kwargs query filters."""
+        try:
+            page = Page.objects.get(
+                ~Q(title_set__publisher_state=PUBLISHER_STATE_PENDING),
+                publisher_is_draft=False,
+                title_set__language=settings.LANGUAGE_CODE,
```

Comment:
Why not use `request.LANGUAGE_CODE`. In this way, we will be ready to enable other languages transparently ?

Reply:
That's a good idea.